src/Controls/src/Core/AnimationExtensions.cs

```diff
@@ -36,7 +37,14 @@ namespace Microsoft.Maui.Controls
 	/// <include file="../../docs/Microsoft.Maui.Controls/AnimationExtensions.xml" path="Type[@FullName='Microsoft.Maui.Controls.AnimationExtensions']/Docs/*" />
 	public static class AnimationExtensions
 	{
-		static readonly Dictionary<int, Animation> s_tweeners;
+		// We use a ConcurrentDictionary because Tweener relies on being able to remove
+		// animations from the AnimationManager within its finalizer (via the Remove extension
+		// method below). Since finalization occurs on a different thread, it risks crashes when
+		// the finalizer is running at the same time another animation is finsihing and removing
+		// itself from this dictionary. So until we can change that design, this dictionary must
+		// be thread-safe.
+		static readonly ConcurrentDictionary<int, Animation> s_tweeners;
+		
 		static readonly Dictionary<AnimatableKey, Info> s_animations;
 		static readonly Dictionary<AnimatableKey, int> s_kinetics;
```

Comment:
These 2 dictionaries are modified at the same time as the tweeners, do they also need to be concurrent?

Reply:
No. They aren't modified from a finalizer. They're only modified on the UI thread by definition.